pkg/acpi/types.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package acpi
+
+// These types are used in emitting binary from a JSON.
+// You can serialize from JSON into structs with these types, and then
+// they can be used in a type switch to serialize out different ways.
+// In one case, the sheap, they serialize into the head and the heap.
+type (
+	// These are serialized as:
+	addr    string // 16 byte IP4 or IP6 addr
+	netaddr string // IP addr as above and 2 byte port
+	flag    string // with other flags in the struct as one u32
```

Comment:
Would it makes more sense for flag to be a bool?

Reply:
will json deserialize into a bool?

Reply:
The whole point of this is that it's something JSON will unmarshall into as strings but I can tell what to do from the type switch.